### PR TITLE
Handle MetaDatum unique but not mandatory

### PR DESCRIPTION
**Pull request by ckaipf**
- This PR updates the validation of the site and submission unique property
- MetaDatum which are unique (and not mandatory) should not evaluate values set to None for unique constraint

### CODE REVIEW

**Comment by ckaipf** [comment]
I think that should only affect MetaDatum which are not mandatory, otherwise we will get an error anyway when value is empty or None

File: datameta/validation.py [hunk 1]
```diff
@@ -135,11 +139,12 @@ def validate_submission_uniquekeys(db, db_files, db_msets):
         for db_mset in db_msets.values():
             for mdatrec in db_mset.metadatumrecords:
                 if mdatrec.metadatum.name == key:
-                    value_msets[mdatrec.value].append(db_mset)
+                    if mdatrec.value is not None:
+                        value_msets[mdatrec.value].append(db_mset)
         # Reduce to those values that occur in more than one metadatast
-        value_msets = { k: v for k, v in value_msets.items() if len(v) > 1 }
+        not_unique = [ v for v in value_msets.values() if len(v) > 1 ]
```
Comment (lkuchenb) [inline datameta/validation.py:145]:
```suggestion
        not_unique = ( v for v in value_msets.values() if len(v) > 1 )
```
No list needed, can use a generator here.

File: datameta/validation.py [hunk 2]
```diff
@@ -135,11 +139,12 @@ def validate_submission_uniquekeys(db, db_files, db_msets):
         for db_mset in db_msets.values():
             for mdatrec in db_mset.metadatumrecords:
                 if mdatrec.metadatum.name == key:
-                    value_msets[mdatrec.value].append(db_mset)
+                    if mdatrec.value is not None:
```
Comment (lkuchenb) [inline datameta/validation.py:142]: Could just be an `and` clause with the previous `if` condition, plain nested `if`s look somewhat funny IMHO

File: datameta/validation.py [hunk 3]
```diff
@@ -148,7 +153,8 @@ def validate_submission_uniquekeys(db, db_files, db_msets):
...
```
Comment (lkuchenb) [inline datameta/validation.py:156]: see above

**Review by lkuchenb — CHANGES_REQUESTED** [review]

**Review by lkuchenb — APPROVED** [review]

**Pull request merged**